packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts

```diff
@@ -585,13 +585,23 @@ export const transformPlotlyJsonToHeatmapProps = (input: PlotlySchema): IHeatMap
     value: 0,
   };
 
-  // Convert normalized values to actual values
-  const domainValuesForColorScale: number[] = firstData.colorscale
-    ? (firstData.colorscale as Array<[number, string]>).map(arr => arr[0] * (zMax - zMin) + zMin)
-    : [];
-  const rangeValuesForColorScale: string[] = firstData.colorscale
-    ? (firstData.colorscale as Array<[number, string]>).map(arr => arr[1])
-    : [];
+  // Initialize domain and range to default values
+  const defaultDomain = [0.0, 0.1, 0.2];
+  const defaultRange = [
+    getColorFromToken(DataVizPalette.success),
+    getColorFromToken(DataVizPalette.warning),
+    getColorFromToken(DataVizPalette.error),
+  ];
+  const domainValuesForColorScale: number[] =
+    firstData.colorscale && firstData.colorscale !== 'Viridis'
```

Comment:
Instead of checking for a specific value, check if the colorscale is of type array or string.